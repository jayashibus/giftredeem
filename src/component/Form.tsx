import React from "react";
import { CONSTANTS } from "../constants";

type IFormProps = {
  validStaff: any;
  staffId: string;
  team_name: string;
  error: string;
  onSubmit: any;
  onChange: (e: any) => void;
};

const Form = ({
  validStaff,
  staffId,
  team_name,
  error,
  onSubmit,
  onChange,
}: IFormProps) => {
  return (
    <form onSubmit={onSubmit}>
      {validStaff ? (
        <>
          <input type="text" value={`ID : ${staffId}`} disabled />
          <input type="text" value={`Team : ${team_name}`} disabled />

          <span className="primary-color">
            {CONSTANTS.giftRedeemScreen.validationMsg.accountVerifiedMsg}
          </span>
        </>
      ) : (
        <>
          <input
            type="text"
            value={staffId}
            placeholder="Enter Staff ID"
            onChange={onChange}
          />
          {error ? <span className="error-message">{error}</span> : ""}
          <button>{CONSTANTS.giftRedeemScreen.button.verify}</button>
        </>
      )}
    </form>
  );
};

export default Form;
