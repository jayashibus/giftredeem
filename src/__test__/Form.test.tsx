import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";

import Form from "../component/Form";

const props = {
  validStaff: true,
  staffId: "STAFF_H123804820G",
  team_name: "RUST",
  error: "",
  onSubmit: () => {},
  onChange: () => {},
};
describe("<Form />", () => {
  test("renders component", () => {
    const { container } = render(<Form {...props} />);
    expect(container).toMatchSnapshot();
  });
});
