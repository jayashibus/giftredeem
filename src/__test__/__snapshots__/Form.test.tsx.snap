// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders component 1`] = `
<div>
  <form>
    <input
      disabled=""
      type="text"
      value="ID : STAFF_H123804820G"
    />
    <input
      disabled=""
      type="text"
      value="Team : RUST"
    />
    <span
      class="primary-color"
    >
      Account Verified Successfully
    </span>
  </form>
</div>
`;
